src/System.Net.Security/src/System/Net/Security/SslStreamPal.Windows.cs

```diff
@@ -131,6 +175,21 @@ public static SafeFreeCredentials AcquireCredentialsHandle(X509Certificate certi
             return AcquireCredentialsHandle(direction, secureCredential);
         }
 
+        internal static byte[] GetNegotiatedApplicationProtocol(SafeDeleteContext context)
+        {
+            Interop.SecPkgContext_ApplicationProtocol alpnContext = SSPIWrapper.QueryContextAttributes(
+                GlobalSSPI.SSPISecureChannel,
+                context,
+                Interop.SspiCli.ContextAttribute.SECPKG_ATTR_APPLICATION_PROTOCOL) as Interop.SecPkgContext_ApplicationProtocol;
+
+            if (alpnContext == null || alpnContext.NegotiationExtension != Interop.ApplicationProtocolNegotiationExt.ALPN || alpnContext.NegotiationStatus != Interop.ApplicationProtocolNegotiationStatus.Success)
```

Comment:
Is there a case where NegotiationExtension isn't ALPN that isn't considered exceptional?

If I ask SChannel for the ALPN data and it gives me back a certificate, I'm certainly upset.

But maybe this is for the ALPN vs NPN thing?  If so, a comment seems warranted.